sched/pthread/pthread_mutex.c

```diff
@@ -38,6 +38,12 @@
 #include "sched/sched.h"
 #include "pthread/pthread.h"
 
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+static spinlock_t g_mutex_spinlock = SP_UNLOCKED;
```

Comment:
move to tcb_s? only need when CONFIG_PTHREAD_MUTEX_UNSAFE isn't true.

Reply:
done

Reply:
need global lock here, for the instance of mutex might be accessed by multiple threads

![image](https://github.com/user-attachments/assets/2d16806c-53a3-4a36-8e05-93dab2c05d46)